### PR TITLE
Remove pointless `JAVA_OPTS` variable

### PR DESCRIPTION
**Pull request by basil**
As far as I can tell, this has been dead code causing nothing but confusion since it was introduced in #1999. I grepped through all the repositories I have checked out in the `jenkinsci` and `apache` GitHub organizations, and while a few `jenkinsci` packaging repositories use `JAVA_OPTS`, nothing consumes it at runtime as far as I can tell. In other words, this is dead code. Perhaps it was added under the mistaken impression that something in the JVM toolchain consumes this, but [this page](https://stackoverflow.com/questions/28327620/difference-between-java-options-java-tool-options-and-java-opts) explains that this is not the case: while there is a `JAVA_TOOL_OPTIONS` variable respected by the JVM, the `JAVA_OPTS` variable is only read by Tomcat and some other applications (like Jenkins packaging code, but _not_ Maven!) that are not invoked in this repository's CI build.

### Testing done

With the changes from https://github.com/jenkinsci/jenkins-test-harness/pull/425, I ran `mvn clean verify -Dtest=hudson.model.ExecutorTest#disconnectCause` and observed that `-XX:InitialHeapSize=523530944 -XX:MaxHeapSize=536870912` was printed. I then ran the same command again after setting `JAVA_OPTS=-Xmx1g -Xms1g` and verified the same `-XX:InitialHeapSize=523530944 -XX:MaxHeapSize=536870912` was printed, proving that setting `JAVA_OPTS` has zero effect and is dead code.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
